### PR TITLE
ANDROAPP-5609-mobile-ui-Create-TEI-Detail-Card-Component

### PR DESCRIPTION
**Pull request by xavimolloy**

### CODE REVIEW

File: designsystem/src/commonMain/resources/drawable/dog.jpg
Comment (xavimolloy) [inline designsystem/src/commonMain/resources/drawable/dog.jpg:1]: file will be deleted after design has been reviewed

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+ * @param additionalInfoList is a list of AdditionalInfoItem that
+ * manages all the key value types that will be shown
+ * if there are more than three items that are not constant
+ * a show more/less button will appear and the rest of items will be hidden
+ * @param expandLabelText the text to be shown for expand button
+ * @param shrinkLabelText the text to be shown for shrink button
+ * @param actionButton composable parameter for the sync button
+ * @param modifier allows a modifier to be passed externally
+ */
+@Composable
```
Comment (xavimolloy) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt:148]: Slightly adapted the Listcard component to fit TEI Detail requirements

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Text.kt [hunk 2]
```diff
@@ -123,3 +126,58 @@ fun Description(
         textAlign = TextAlign.Start,
     )
 }
+
+@Composable
+fun ListCardTitle(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Text.kt:131]: should this be internal?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Text.kt
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Text.kt:1]: should all these new methods be internal?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+fun ListCardDetail(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt:149]: The previous function is intended to render list items so it's called `ListCard`, as this function is intended to render TEI details, maybe we can remove the List and call it just `CardDetail`. What do you think?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+ *  used to display the avatar composable in card,
+ *  must be one of the three styles given as parameters
+ * @param style not nullable parameter that manages the avatar style
+ * @param textAvatar style must be TEXT, will show a single character as avatar
+ * @param imagePainter style must be IMAGE, will display an image as avatar
+ * @param metadataAvatar style must be METADATA, composable should be DHIS2 [MetadataAvatar]
+ * @param modifier allows a modifier to be passed externally
+ */
+@Composable
+fun ListAvatar(
```
Comment (andresmr) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt:224]: I think this could be named just Avatar

**Review by andresmr — APPROVED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Text.kt (hunk 2 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/Text.kt:131]: yes it should I'm touching up some things so I'll add this in

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt (hunk 3 above)
Comment (xavimolloy) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt:149]: I agree naming could be improved

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt [hunk 5]
```diff
@@ -207,19 +288,19 @@ private fun AdditionalInfoColumn(
     ) {
         if (expandableItems != null && expandableItems.size > 3) {
             expandableItemList = expandableItems.take(3).toMutableList()
-            KeyValueList(expandableItemList)
+            KeyValueList(expandableItemList, isTEIDetailItem = isTEIDetailItem)
```
Comment (DavidAparicioAlbaAsenjo) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ListCard.kt:291]: the naming it's a little confusing

**Review by andresmr — APPROVED** [review]

**Pull request merged**